gammapy/detect/cwt.py

```diff
@@ -57,200 +58,334 @@ def difference_of_gauss_kernel(radius, scale_step, n_sigmas=8):
+            self._execute_iteration(data=data)
+            if self._is_converged(data=data):
+                break
+
+        if n_iter != self.max_iter:
+            log.info('Convergence reached at iteration {0}'.format(n_iter + 1))
+        else:
+            log.info('Convergence not formally reached at iteration {0}'.format(n_iter + 1))
+
+
+class CWTData(object):
+    """
+    Images for CWT algorithm.
+
+    Parameters
+    ----------
+    counts : `~numpy.ndarray`
+        2D counts image.
+    background : `~numpy.ndarray`
+        2D background image.
+
+    Attributes
+    ----------
+    counts : `~numpy.ndarray`
+        2D counts input image.
+    background : `~numpy.ndarray`
+        2D background input image.
+    model : `~numpy.ndarray`
+        2D image. Primordial initialized by zero array.
+    approx : `~numpy.ndarray`
+        2D image. Primordial initialized by zero array. In the course of
+        iterations updated by convolution of ``counts - model - background``
+        with ``kern_approx``
+    approx_bkg : `~numpy.ndarray`
+        2D image. Primordial initialized by zero array. In the course of
+        iterations updated by convolution of ``background`` with ``kern_approx``
+    transform : `~numpy.ndarray`
+        3D cube. Primordial initialized by None. In the course of
+        iterations updated by convolution of ``counts - total_background`` with kernel
+        for each scale (``total_background = model + background + approx``).
+    error : `~numpy.ndarray`
+        3D cube. Primordial initialized by None. In the course of
+        iterations updated by convolution of ``total_background`` with kernel^2
+        for each scale
+    support : `~numpy.ndarray`
+        3D cube exclusion mask. Primordial initialized by None.
+    mysterious_result : `~numpy.ndarray`
+        2D image. Primordial initialized by None.
+
+    Examples
+    --------
+    >>> import os
+    >>> from astropy.io import fits
+    >>> from gammapy.detect import CWTData
+    >>> filename = os.environ['GAMMAPY_EXTRA'] + '/datasets/fermi_survey/all.fits.gz'
+    >>> image = fits.getdata(filename, hdu='COUNTS')
+    >>> background = fits.getdata(filename, hdu='BACKGROUND')
+    >>> data = CWTData(counts=image, background=background)
+    >>> print (data)
+    <gammapy.detect.cwt.CWTData object at 0x7fccb6acc990>
+    """
+
+    def __init__(self, counts, background):
+        self.counts = np.array(counts, dtype=float)
+        self.background = np.array(background, dtype=float)
+
+        shape_2d = self.counts.shape
+        self.model = np.zeros(shape_2d)
+        self.approx = np.zeros(shape_2d)
+        self.approx_bkg = np.zeros(shape_2d)
+
+        self.transform = None
+        self.error = None
+        self.support = None
+
+        # Strange result after each iteration of CWT...
+        self.mysterious_result = None
 
     @property
     def max_scale_image(self):
```

Comment:
@cdeil I moved this method here, but as you can see it will be not work, because we don't have some information about the scales in that class